codegen/generator/templates/src/object.go.tmpl

```diff
@@ -33,30 +33,30 @@ type {{ $field | FieldOptionsStructName }} struct {
 	{{- end }}
 	{{- end }}
 	{{ if $field.TypeRef.IsObject }}
-	return &{{ $field.TypeRef | FormatType }} {
+	return &{{ $field.TypeRef | FormatOutputType }} {
 		q: q,
 		c: r.c,
 	}
 	{{- else if or $field.TypeRef.IsScalar $field.TypeRef.IsList }}
-	var response {{ $field.TypeRef | FormatType }}
+	var response {{ $field.TypeRef | FormatOutputType }}
 	q = q.Bind(&response)
 	return response, q.Execute(ctx, r.c)
 	{{- end }}
 }
 
 {{ if eq $field.Name "id" }}
-// graphqlType returns the native GraphQL type name
-func (r *{{ $.Name | FormatName }}) graphqlType() string {
+// XXGraphqlType is an internal function. It returns the native GraphQL type name
+func (r *{{ $.Name | FormatName }}) XXGraphQLType() string {
 	return "{{ $.Name }}"
 }
 
-// graphqlMarshal serializes the structure into GraphQL
-func (r *{{ $.Name | FormatName }}) graphqlMarshal(ctx context.Context) (any, error) {
+// XXGraphqlID is an internal function. It returns the underlying type ID
+func (r *{{ $.Name | FormatName }}) XXGraphQLID(ctx context.Context) (string, error) {
```

Comment:
Guessing these will show up in the SDK godoc now?

Reply:
Unfortunately, yes :/

Problem stems from these being defined in the root package (e.g. api.gen.go) and then used in `internal/querybuilder` -- can't use private functions there.

Unless there's a way out?

I've used the (crappy) `XX` convention to somewhat make it explicit in code completion that they're not to be toyed with.

Perhaps we should use gRPC convention (`XXX_Whatever`) or make them part of the public API by having more generic name/meaning

Reply:
Hm yeah unfortunate but not a dealbreaker, makes sense why it's needed.

The only idea I have is to move it to an embedded type, which I think will keep it from showing up in the godoc (and MAYBE we can even make the type private...?), but it has a dependency on `r` for `r.ID` so that won't help. Unless it just has its own `ID(ctx)` implementation somehow.

Failing that it might be worth following the gRPC/protobuf convention, just because, but not a blocker.